### PR TITLE
[Monitoring] Add auto cancel script

### PR DESCRIPTION
**Pull request by AyushR1**

### CODE REVIEW

File: scripts/monitoring/auto_cancel_submissions.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        submissions = evalai.get_submissions_for_challenge(
+            challenge_pk, "finished"
+        )
+        submissions += evalai.get_submissions_for_challenge(
+            challenge_pk, "running"
+        )
+        submissions += evalai.get_submissions_for_challenge(
+            challenge_pk, "resuming"
+        )
+        print(submissions)
```
Comment (gchhablani) [inline scripts/monitoring/auto_cancel_submissions.py:38]: Remove this print statement.

File: scripts/monitoring/auto_cancel_submissions.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        current_time = datetime.now(pytz.utc)
+        for submission in submissions:
+            status = submission["status"]
+            submission_time = get_submission_time(submission)
+            submission_time = pytz.utc.localize(submission_time)
+
+            time_difference = current_time - submission_time
+            if time_difference > timedelta(days=days_threshold):
+                data = {"status": "cancelled"}
+                evalai.update_submission_status(data, submission["challenge"])
```
Comment (gchhablani) [inline scripts/monitoring/auto_cancel_submissions.py:49]: What does `submission["challenge"]` return?

Comment (gchhablani) [reply scripts/monitoring/auto_cancel_submissions.py:49]: Are you sure this is correct?

File: scripts/monitoring/evalai_interface.py [hunk 3]
```diff
@@ -119,8 +132,10 @@ def get_challenge_by_pk(self, challenge_pk):
         response = self.make_request(url, "GET")
         return response
 
-    def get_challenges(self):
+    def get_challenges(self, next_page_url=None):
         url = URLS.get("get_challenges")
         url = self.return_url_per_environment(url)
+        if next_page_url:
+            url = next_page_url
```
Comment (gchhablani) [inline scripts/monitoring/evalai_interface.py:139]: This is not needed. Add a while loop in the auto-cancel script just like other scripts.

File: tests/unit/challenges/test_views.py [hunk 4]
```diff
@@ -4096,6 +4096,7 @@ def test_get_all_submissions_when_user_is_host_of_challenge(self):
                     "submitted_at": "{0}{1}".format(
                         submission.submitted_at.isoformat(), "Z"
                     ).replace("+00:00", ""),
+                    "rerun_resumed_at": None,
```
Comment (gchhablani) [inline tests/unit/challenges/test_views.py:4099]: These are all wrong.
Use `self.submission<x>.rerun_resumed_at` following others.

**Review by gchhablani — COMMENTED** [review]
Added some comments.

File: tests/unit/challenges/test_views.py (hunk 4 above)
Comment (gchhablani) [reply tests/unit/challenges/test_views.py:4099]: Same as `started_at` etc.

File: scripts/monitoring/evalai_interface.py (hunk 3 above)
Comment (ayushr1-zk) [reply scripts/monitoring/evalai_interface.py:139]: I will need to write the request function then here only as the url is passed with a param for the next page. The other places have there own request function written and doesn't really use the eval interface

**Comment by gchhablani** [comment]
@AyushR1 Separate the addition of the new attribute and the script in two PRs. Keep this one only for auto-cancel.

File: scripts/monitoring/evalai_interface.py (hunk 3 above)
Comment (gchhablani) [reply scripts/monitoring/evalai_interface.py:139]: @AyushR1 No. Have you checked EvalAI Interface? There is a `make_request` function that you can use.

**Comment by gchhablani** [comment]
@AyushR1 There are other APIs to update submissions.

**Pull request closed without merging**